packages/batch-submitter/src/exec/run-batch-submitter.ts

```diff
@@ -0,0 +1,102 @@
+/* External Imports */
+import { getLogger } from '@eth-optimism/core-utils'
+import { exit } from 'process'
+import {
+  getContractInterface,
+  getContractFactory,
+} from '@eth-optimism/contracts'
+import { Signer, Wallet } from 'ethers'
+import { Provider, JsonRpcProvider } from '@ethersproject/providers'
+import { OptimismProvider } from '@eth-optimism/provider'
+
+/* Internal Imports */
+import { BatchSubmitter, CanonicalTransactionChainContract } from '..'
+
+/* Logger */
+const log = getLogger('oe:batch-submitter:init')
+
+interface RequiredEnvVars {
+  ADDRESS_MANAGER_ADDRESS: 'ADDRESS_MANAGER_ADDRESS'
+  SEQUENCER_PRIVATE_KEY: 'SEQUENCER_PRIVATE_KEY'
+  INFURA_NETWORK: 'INFURA_NETWORK'
+  INFURA_PROJECT_ID: 'INFURA_PROJECT_ID'
+  L2_WEB3_URL: 'L2_WEB3_URL'
+  L2_CHAIN_ID: 'L2_CHAIN_ID'
+  MAX_TX_SIZE: 'MAX_TX_SIZE'
+  POLL_INTERVAL: 'POLL_INTERVAL'
+  DEFAULT_BATCH_SIZE: 'DEFAULT_BATCH_SIZE'
+  NUM_CONFIRMATIONS: 'NUM_CONFIRMATIONS'
+}
+const requiredEnvVars: RequiredEnvVars = {
+  ADDRESS_MANAGER_ADDRESS: 'ADDRESS_MANAGER_ADDRESS',
+  SEQUENCER_PRIVATE_KEY: 'SEQUENCER_PRIVATE_KEY',
+  INFURA_NETWORK: 'INFURA_NETWORK',
+  INFURA_PROJECT_ID: 'INFURA_PROJECT_ID',
+  L2_WEB3_URL: 'L2_WEB3_URL',
+  L2_CHAIN_ID: 'L2_CHAIN_ID',
+  MAX_TX_SIZE: 'MAX_TX_SIZE',
+  POLL_INTERVAL: 'POLL_INTERVAL',
+  DEFAULT_BATCH_SIZE: 'DEFAULT_BATCH_SIZE',
+  NUM_CONFIRMATIONS: 'NUM_CONFIRMATIONS',
+}
+
+export const run = async () => {
+  log.info('Starting batch submitter...')
+
+  for (const val in requiredEnvVars) {
+    if (!process.env[val]) {
+      log.error(`Missing enviornment variable: ${val}`)
+      exit(1)
+    }
+  }
+  Object.assign(requiredEnvVars, process.env)
```

Comment:
This seems like it brings in all env vars instead of just the required ones?

Reply:
Oh yeah true. Is it unclean to bring in all the env vars even though I won't use them? Is there an easy way to filter them?